pytorch_lightning/trainer/attributes.py

```diff
@@ -0,0 +1,29 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC
+from typing import Any, Dict, List, Optional, Union
+
+
+class TrainerAttributes(ABC):
```

Comment:
We are moving away from Mixin as hard to track and debug.

Reply:
What do you suggest for declaring attributes then? Directly on the `Trainer` class?